ElectronClient/app.js

```diff
@@ -1166,21 +1169,23 @@ class Application extends BaseApplication {
 		this.lastMenuScreen_ = screen;
 	}
 
-	async updateMenuItemStates(state = null) {
+	async updateMenuItemStates(viewer, state = null) {
 		if (!this.lastMenuScreen_) return;
 		if (!this.store() && !state) return;
 
 		if (!state) state = this.store().getState();
 
 		const selectedNoteIds = state.selectedNoteIds;
 		const note = selectedNoteIds.length === 1 ? await Note.load(selectedNoteIds[0]) : null;
-
 		for (const itemId of ['copy', 'paste', 'cut', 'selectAll', 'bold', 'italic', 'link', 'code', 'insertDateTime', 'commandStartExternalEditing', 'showLocalSearch']) {
 			const menuItem = Menu.getApplicationMenu().getMenuItemById(`edit:${itemId}`);
-			if (!menuItem) continue;
-			menuItem.enabled = !!note && note.markup_language === MarkupToHtml.MARKUP_LANGUAGE_MARKDOWN;
+			const noteCheck = !!note && note.markup_language === MarkupToHtml.MARKUP_LANGUAGE_MARKDOWN;
+			if (noteCheck === false || viewer === 'viewer') {
+				menuItem.enabled = false;
+			} else if (noteCheck === true || viewer !== 'viewer') {
+				menuItem.enabled = true;
+			}
```

Comment:
Sorry I can't wrap my head around this logic. There's an implied "else" case - in which case is it executed? Never? Then why use an else/if?

Also please could you clarify what "noteCheck" means? Is it possible to find a name that better expresses your intent?

Reply:
about the else/if statement, i have changed that to else and the logic still works fcine. noting was broken.

and the `noteCheck` variable is used to store the boolean value that will be returned from this statement `!note && note.markup_language === MarkupToHtml.MARKUP_LANGUAGE_MARKDOWN;`, which is used to check if there is a note and also if the note is a HTML note.

i have added a comment to explain that part.

Reply:
Hi @Runo-saduwa, I don't want to interfere, but just to suggest maybe something like 'isHtmlNote' might be a better name for 'noteCheck'.  Or ideally maybe don't bother with making a temporary variable at all because you don't need it again.

Another style tip, consider using just 'if (x)' instead of if (x === true)' unless you have an explicit reason for doing the latter. Writing it this way will also help you think of a good name for it.

Reply:
@mic704b that’s a great suggestion, thanks. I’ll with your suggestions. 

isHtmlNote is cool

Reply:
The code has been improved again, this time I improved the variable name to something meaningful (isHtmlNote) and I also replaced  the If statement with a sweet ternary operator. 

You can go ahead and test it, noting is broken, everything is in order.

Reply:
>  I also replaced the If statement with a sweet ternary operator

Better, but switch it around and it would be even better.

Reply:
And remove the comment about `noteCheck`

Reply:
Yeah Thanks, I just did that